### PR TITLE
Run QEMU tests with 2 CPUs (and add a flag to add more)

### PR DESCRIPTION
**Pull request by andrisaar**
We've done a lot of work to enable multiple cores in Oak Containers; let's run the tests with multiple CPUs as well.

Ref #4235

### CODE REVIEW

File: oak_containers_launcher/src/qemu.rs [hunk 1]
```diff
@@ -50,6 +50,10 @@ pub struct Params {
     #[arg(long)]
     pub memory_size: Option<String>,
 
+    /// How many CPUs to give to the VM.
```
Comment (mariaschett) [inline oak_containers_launcher/src/qemu.rs:53]: Haha, what happens when `num_cpus` is None? No CPUs for the VM? Maybe 1 as default value? And maybe add in a comment what the largest number is you'd expect work/tested with?

File: oak_containers_launcher/src/qemu.rs [hunk 2]
```diff
@@ -109,6 +114,9 @@ impl Qemu {
         if let Some(memory_size) = params.memory_size {
             cmd.args(["-m", &memory_size]);
         };
+        if let Some(num_cpus) = params.num_cpus {
```
Comment (mariaschett) [inline oak_containers_launcher/src/qemu.rs:117]: Ah, I see now why you have Some, but I'd assume it would work with 1 too and you could drop the if?

**Review by mariaschett — APPROVED** [review]
LGTM, thank you!

File: oak_containers_launcher/src/qemu.rs (hunk 2 above)
Comment (andrisaar) [reply oak_containers_launcher/src/qemu.rs]: Done; I've changed it to use a default value of 1.

**Pull request merged**